src/adapters/envista.js

```diff
@@ -1,482 +1,167 @@
 
-export const name = 'envista';
+export const name = 'envista-new';
 
 export function fetchData (source, cb) {
-  let menuSiteUrl = source.url + 'MenuSite.aspx';
-  request(source)(menuSiteUrl, (err, res, body) => {
+  let regionListUrl = source.url + 'regions';
+  requestHeaders(regionListUrl, (err, res, body) => {
     if (err || res.statusCode !== 200) {
       return cb({message: 'Failure to load data url.'});
     }
 
     let tasks = [];
-    let cityLinks = body.match(/DynamicTable.aspx\?G_ID=(\d*)/g);
-    if (!cityLinks) {
-      return cb('Unable to match cities', []);
-    }
-    while (cityLinks.length > 0) {
-      let link = source.url + cityLinks.pop();
-      tasks.push(handleCity(source, link));
-    }
+    const regionList = JSON.parse(body);
+    regionList.forEach(region => {
+      tasks.push(handleRegion(source, region));
+    });
 
     parallel(tasks, (err, results) => {
       if (err) {
         return cb(err, []);
       }
-
       results = flattenDeep(results);
       results = convertUnits(results);
-
       return cb(err, {name: 'unused', measurements: results});
     });
   });
 }
 
-const request = function (source) {
-  switch (source.country) {
-    case 'ZA':
-      return requestDefault;
-    case 'IL':
-      return requestJarAndHeaders;
-    default:
-      return requestDefault;
-  }
-};
-
-const handleCity = function (source, link) {
+const handleRegion = function (source, region) {
+  let stationList = region.stations;
   return function (done) {
-    request(source)(link, (err, res, body) => {
-      if (err || res.statusCode !== 200) {
-        return done(null, []);
-      }
-
-      let tasks = [];
-      let stationURLs = body.match(/StationInfo[5]?.aspx\?ST_ID=(\d*)/g);
-      const $ = cheerio.load(body);
-      const regionName = $('#lblCaption').text().split('-')[1].trim();
-      while (stationURLs.length > 0) {
-        const stationURL = stationURLs.pop();
-        const stationID = stationURL.match(/StationInfo5?.aspx\?ST_ID=(\d*)/)[1];
-        let link = source.url + stationURL;
-        link = link.replace(/StationInfo[5]?/, 'StationReportFast');
-        tasks.push(handleStation(source, link, regionName, stationID));
+    let tasks = [];
+    stationList.forEach(station => {
+      if (station.active && hasAcceptedParameters(station)) {
+        tasks.push(handleStation(source, region.name, station));
       }
-
-      parallelLimit(tasks, 3, (err, results) => {
-        return done(err, results);
-      });
     });
-  };
-};
-
-// station query page
-const handleStation = function (source, link, regionName, stationID) {
-  return function (done) {
-    request(source)(link, (err, res, body) => {
-      if (err || res.statusCode !== 200) {
-        return done(null, []);
-      }
 
-      const $ = cheerio.load(body);
-      if (!hasAcceptedParameters($)) {
-        return done(null, []);
+    parallelLimit(tasks, 16, (err, results) => { //  Magic number to avoid rate limiting is 16.
```

Comment:
Does 17 break it? would going down to 15 break anything else. Might be better safe than sorry

Reply:
I'll reduce it to 15 to be safe 👍🏼